### ISSUE TITLE
Multiple scores in searchlight

### ISSUE THREAD

**Issue by robbisg** [nilearn/nilearn#1632]
Hi,

I am implementing (almost finished) the possibility to estimate multiple scoring functions for the searchlight analysis, using sklearn cross_validate function and returning a dictionary of maps.

I can PR the code, but I don't want to break up tests and so on, so if you are interested I will need some infos to correctly perform a smooth PR! ;)

**Comment by bthirion** [comment]
Well there are 2 things:
* First check that the code is not broken after you change. For that you need to run nostests.
* Second, if you added a new functionality, you should test it in particular.

More indications here:  http://scikit-learn.org/stable/developers/contributing.html#coding-guidelines

**Comment by robbisg** [comment]
Hi,

I've implemented the feature, modified tests and build another little test for the new feature!

**Comment by NicolasGensollen** [comment]
Hi @robbisg 
Any updates on this or should I close the issue?

**Comment by robbisg** [comment]
Hi @NicolasGensollen ,

I did this feature several time ago and it should be tested with the newest version of the code, but I don't know if is ok to include this in nilearn. If yes, I can PR otherwise it is ok to close!

Thank you!

**Comment by NicolasGensollen** [comment]
@robbisg you're welcome to open a PR which will make it easy to assess how your new feature works and how it could be integrated in the newest version of the code. Let me know if I can help!

**Comment by robbisg** [comment]
Great! I hope to open it asap! Thanks @NicolasGensollen !

**Comment by robbisg** [comment]
@NicolasGensollen @bthirion @GaelVaroquaux 
I was modifying this feature and I found that in master version cross-validation results are averaged, but I use to keep these results since sometimes these may be useful (e.g. across-subjects analyses or check the variability of accuracy wrt selected cv-schema). 
Do you think it is useful to keep cv fold values? This may impact also some examples.

**Comment by bthirion** [comment]
In general, I agree with you that having an eye on accuracy variability is instructive !
In the case of searchlight-type analysis, this information is a bit hard to represent as it is varies from voxel to voxel. You cannot do a boxplot etc. How would you convey it to the user ?

**Comment by robbisg** [comment]
I was only wondering if you think that providing also the each cross-validation searchlight map could be useful. since now in ``` _group_iter_search_light``` line 171 is averaging across cv-folds.
``` python
par_scores[i] = np.mean(cross_val_score(estimator, X[:, row],
                                        y, cv=cv, n_jobs=1, **kwargs))
```
I removed the mean and now ```sl.scores_``` is a 4D image with the fourth dimension representing the cv-fold, then the users can use this information as they want and this new output will impact also searchlight examples that should be modified accordingly, but I wanted to know, before opening the PR, if you think that this new output is useful.

**Comment by bthirion** [comment]
I think I'm fine with it, but I don't use searchlight, so we should have somebody else's opinion.
@SylvainTakerkart ?

**Comment by SylvainTakerkart** [comment]
Hi, I agree: being able to have access to single-fold accuracy maps is very useful, in particular to perform stats on searchlight maps. I've had to use various tricks in the past for this, and the one you propose, @robbisg (remove the np.mean on that line), seems like a good simple one!
